hexl/thread-pool/thread-pool-vars-util.hpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <math.h>
+
+#include <cstdlib>
+#include <string>
+#include <thread>
+
+#include "hexl/logging/logging.hpp"
+#include "hexl/thread-pool/thread-pool-vars.hpp"
+
+namespace intel {
+namespace hexl {
+
+const uint64_t HEXL_DEFAULT_NUM_THREADS = 16;
+const uint64_t HEXL_DEFAULT_NTT_PARALLEL_DEPTH = 1;
+
+#ifdef HEXL_MULTI_THREADING
+
+// Check for environment variable
+static int check_env_var(const char* var) {
+  int value = 0;
+
+  // Get value from env variable
+  char* var_value = std::getenv(var);
+  if (var_value != NULL) {
+    value = static_cast<int>(std::strtol(var_value, NULL, 10));
+  }
```

Comment:
```suggestion
  if (var_value != nullptr) {
    value = static_cast<int>(std::strtol(var_value, nullptr, 10));
  }
```
Never any need in c++ to use NULL macro